gammapy/spectrum/dataset.py

```diff
@@ -676,6 +677,96 @@ def read(cls, filename):
             "To read from an OGIP fits file use SpectrumDatasetOnOff.from_ogip_files."
         )
 
+    def stack(self, other):
+        """Stack this dataset with another one.
+
+        Safe mask is applied to compute the stacked counts vector.
+        Counts outside each dataset safe mask are lost.
+
+        Stacking is performed in-place.
+
+        Parameters
+        ----------
+        other : `~gammapy.spectrum.SpectrumDatasetOnOff`
+            the dataset to stack to the current one
+       """
+
+        def is_valid(dataset):
+            if (
+                dataset.acceptance_off is None
+                or dataset.acceptance is None
+                or dataset.counts_off is None
+            ):
+                return False
+            else:
+                return True
+
+        if not isinstance(other, SpectrumDatasetOnOff):
+            raise TypeError("Incompatible types for SpectrumDatasetOnOff stacking")
+
+        # We assume here that counts_off, acceptance and acceptance_off are well defined.
+        if not is_valid(self) or not is_valid(other):
+            raise ValueError("Cannot stack incomplete SpectrumDatsetOnOff.")
+
+        total_off = np.zeros_like(self.counts_off.data, dtype=float)
+        total_alpha = np.zeros_like(self.counts_off.data, dtype=float)
+
+        total_off[self.mask_safe] += self.counts_off.data[self.mask_safe]
+        total_off[other.mask_safe] += other.counts_off.data[other.mask_safe]
+        total_alpha[self.mask_safe] += (self.alpha * self.counts_off)[self.mask_safe]
+        total_alpha[other.mask_safe] += (other.alpha * other.counts_off)[
+            other.mask_safe
+        ]
+
+        with np.errstate(divide="ignore", invalid="ignore"):
+            acceptance_off = total_off / total_alpha
+            average_alpha = total_alpha.sum() / total_off.sum()
+
+        acceptance = np.ones_like(self.counts_off.data, dtype=float)
+        idx = np.where(total_off == 0)[0]
+        # For the bins where the stacked OFF counts equal 0, the alpha value is performed by weighting on the total
+        # OFF counts of each run
+        acceptance_off[idx] = 1 / average_alpha
+
+        self.acceptance = acceptance
```

Comment:
Actually the stacked one is forced to be 1. So it can be different from the input one.

Reply:
OK, that make sense. I think so far we only use `acceptance=1` and `acceptance_off=1 / alpha` in Gammapy, but I agree it might change and be too limiting...